src/OcisResource.php

```diff
@@ -55,6 +56,24 @@ private function getMetadata(ResourceMetadata $property): array|string
         return $metadata[$property->getKey()];
     }
 
+    /**
+     * gets all possible permissions for the resource
```

Comment:
```suggestion
     * gets all possible roles for the resource
```

Is this intended to return all the roles that the current user has for the resource? (including any "sub-roles" that have a subset of the user's permissions.)
And the assumption is that the current user will be able to "pass on" (invite) to another user any of the returned roles when they "invite" (share) to some other user.

Reply:
This function will get all (sharing)roles that a specific user can set for this resource. 
`"@libre.graph.permissions.roles.allowedValues"` part of https://owncloud.dev/libre-graph-api/#/drives.permissions/ListPermissions